spec/text-editor-spec.coffee

```diff
@@ -1623,7 +1623,56 @@ describe "TextEditor", ->
             [[6, 22], [6, 28]]
           ]
 
+        it "selects also soft-wrapped lines", ->
+          editor.setSoftWrapped(true)
+          editor.setDefaultCharWidth(10)
+          editor.setEditorWidthInChars(40)
+
+          editor.setSelectedScreenRange([[6, 20], [6, 25]])
+          editor.addSelectionBelow()
+          expect(editor.getSelectedScreenRanges()).toEqual [
+            [[6, 20], [6, 25]]
+            [[7, 20], [7, 25]]
+          ]
+
+        it "takes atomic tokens into account", ->
+          waitsForPromise ->
+            atom.project.open('sample-with-tabs-and-leading-comment.coffee', autoIndent: false).then (o) -> editor = o
+
+          runs ->
+            editor.setSelectedBufferRange([[2, 1], [2, 3]])
+            editor.addSelectionBelow()
+
+            expect(editor.getSelectedBufferRanges()).toEqual [
+              [[2, 1], [2, 3]]
+              [[3, 1], [3, 2]]
+            ]
+
       describe "when the selection is empty", ->
+        describe "when lines are soft-wrapped", ->
+          beforeEach ->
+            editor.setSoftWrapped(true)
+            editor.setDefaultCharWidth(10)
```

Comment:
Setting the default character width is actually messing up the subsequent call to `setEditorWidthInChars` and causing it not to work. So basically none of the lines are actually soft-wrapping in this test.